test/src/test/java/jenkins/triggers/ReverseBuildTriggerTest.java

```diff
@@ -152,6 +152,41 @@
         r.waitUntilNoActivity();
         assertEquals(3, downstream.getLastBuild().number);
         assertEquals(new Cause.UpstreamCause((Run) b), downstream.getLastBuild().getCause(Cause.UpstreamCause.class));
+
+        // Alice can only DISCOVER upstream, so downstream does not get built, but the upstream build cannot DISCOVER downstream
+        perms = new HashMap<Permission,Set<String>>();
+        perms.put(Item.DISCOVER, new HashSet<String>(Arrays.asList("alice")));
+        perms.put(Item.READ, new HashSet<String>(Arrays.asList("bob")));
+        upstream.removeProperty(AuthorizationMatrixProperty.class);
+        upstream.addProperty(new AuthorizationMatrixProperty(perms));
```

Comment:
Oof, this should all be using `MockAuthorizationStrategy` by now for readability…though I had already made that change in this test but apparently not.